misc/python/materialize/data_ingest/executor.py

```diff
@@ -0,0 +1,359 @@
+        self.conn = pg8000.connect(
+            host="localhost",
+            user="postgres",
+            password="postgres",
+            port=ports["postgres"],
+        )
+        self.table = f"table{num}"
+        self.fields = fields
+
+        values = [
+            f"{field.name} {str(field.typ.name(Backend.POSTGRES)).lower()}"
+            for field in fields
+        ]
+        keys = [field.name for field in fields if field.is_key]
+
+        with self.conn.cursor() as cur:
+            cur.execute(
+                f"""DROP TABLE IF EXISTS {self.table};
+                    CREATE TABLE {self.table} (
+                        {", ".join(values)},
+                        PRIMARY KEY ({", ".join(keys)}))"""
+            )
+
+    def run(self, transactions: List[Transaction]) -> None:
+        for i, transaction in enumerate(transactions):
+            self.print_progress(i, len(transactions))
+            with self.conn.cursor() as cur:
+                for row_list in transaction.row_lists:
+                    for row in row_list.rows:
+                        if (
+                            row.operation == Operation.INSERT
+                            or row.operation == Operation.UPSERT
+                        ):
+                            values_str = ", ".join(
+                                str(field.formatted_value()) for field in row.fields
+                            )
+                            keys_str = ", ".join(
+                                field.name for field in row.fields if field.is_key
+                            )
+                            update_str = ", ".join(
+                                f"{field.name} = EXCLUDED.{field.name}"
+                                for field in row.fields
+                                if not field.is_key
+                            )
+                            cur.execute(
+                                f"""INSERT INTO {self.table}
+                                    VALUES ({values_str})
+                                    ON CONFLICT ({keys_str})
+                                    DO UPDATE SET {update_str}
+                                """
+                            )
+                        elif row.operation == Operation.DELETE:
+                            cond_str = " AND ".join(
+                                f"{field.name} = {field.formatted_value()}"
+                                for field in row.fields
+                                if field.is_key
+                            )
+                            cur.execute(
+                                f"""DELETE FROM {self.table}
+                                    WHERE {cond_str}
+                                """
+                            )
+                        else:
+                            raise ValueError(f"Unexpected operation {row.operation}")
+        self.conn.commit()
+
+
+class PgCdcExecutor(Executor):
```

Comment:
I think this mostly duplicates the `PgExecutor` above. Why do you need two Postgres executors?